# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.cpp
# Commit: 05896dbeb434
# Full Hash: 05896dbeb434d6f3ca9324f01eb67020d932f50f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Determine webgl's layers::TextureType in Present. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79617
# ==============================================================================

diff -r 8439f89e170b -r 05896dbeb434 dom/canvas/WebGLContext.cpp
--- a/dom/canvas/WebGLContext.cpp	Mon Jun 15 18:26:12 2020 +0000
+++ b/dom/canvas/WebGLContext.cpp	Mon Jun 15 18:26:14 2020 +0000
@@ -1006,17 +1006,11 @@
 }
 
 Maybe<layers::SurfaceDescriptor> WebGLContext::GetFrontBuffer(
-    WebGLFramebuffer* const fb, const layers::TextureType consumerType) {
+    WebGLFramebuffer* const fb) {
   auto swapChain = &mSwapChain;
   if (fb) {
     swapChain = &fb->mOpaqueSwapChain;
   }
-
-  if (swapChain->mFactory &&
-      swapChain->mFactory->mDesc.consumerType != consumerType) {
-    swapChain->mFactory = nullptr;  // Better luck next Present.
-  }
-
   const auto& front = swapChain->FrontBuffer();
   if (!front) return {};
 